orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -843,3 +853,357 @@ def _update_onnx_model_initializers(self, state_tensors):
         for w_i in replace_indices:
             del self._onnx_model.graph.initializer[w_i]
         self._onnx_model.graph.initializer.extend(new_weights)
+
+    def _extract_model_states(self, state_dict, pytorch_format):
+        """Extract model states from the training session and load into the state_dict"""
+
+        model_states = self._training_session.get_model_state(include_mixed_precision_weights=False)
+        state_dict[_utils.state_dict_model_key()] = {}
+
+        # extract trained model weights from the training session
+        for precision in model_states:
+            state_dict[_utils.state_dict_model_key()][precision] = {}
+            for model_state_key in model_states[precision]:
+                if pytorch_format:
+                    state_dict[_utils.state_dict_model_key()][precision][model_state_key] = \
+                        torch.from_numpy(model_states[precision][model_state_key])
+                else:
+                    state_dict[_utils.state_dict_model_key()][precision][model_state_key] = \
+                        model_states[precision][model_state_key]
+
+        # extract untrained (frozen) model weights
+        for node in self._onnx_model.graph.initializer:
+            if node.name not in state_dict[_utils.state_dict_model_key()][_utils.state_dict_full_precision_key()] and \
+                node.name in self.options.utils.frozen_weights:
+                if pytorch_format:
+                    state_dict[_utils.state_dict_model_key()][_utils.state_dict_full_precision_key()][node.name] = \
+                        torch.from_numpy(onnx.numpy_helper.to_array(node))
+                else:
+                    state_dict[_utils.state_dict_model_key()][_utils.state_dict_full_precision_key()][node.name] = \
+                        onnx.numpy_helper.to_array(node)
+
+    def _extract_trainer_options(self, state_dict):
+        """Extract relevant trainer configuration and load it into the state_dict"""
+
+        state_dict[_utils.state_dict_trainer_options_key()] = {}
+        state_dict[_utils.state_dict_trainer_options_key()]['mixed_precision'] = self.options.mixed_precision.enabled
+        state_dict[_utils.state_dict_trainer_options_key()]['zero_stage'] = \
+            self.options.distributed.deepspeed_zero_optimization.stage or 0
+        state_dict[_utils.state_dict_trainer_options_key()]['world_rank'] = self.options.distributed.world_rank or 0
```

Comment:
self.options.distributed.world_rank already defaults to 0, so the `or 0` shouldn't be needed, right? Same for the world_size that defaults to 1 and zero which defaults to 0

Reply:
Ok, that sounds good. I can remove the or part here.